design-proposals/virt-operator-deploy-instancetypes.md

```diff
@@ -0,0 +1,139 @@
+[specific release](https://github.com/kubevirt/common-instancetypes/releases/tag/v0.3.1)
+or by using the [SSP operator](https://github.com/kubevirt/ssp-operator#ssp-operator).
+
+This proposal suggests moving the deployment of the cluster-wide
+`kubevirt/common-instancetypes` resources to `virt-operator` so that they are
+consistently available across KubeVirt deployments.
+
+## Motivation
+
+The ultimate goal of this proposal is to improve KubeVirt's default user
+experience when creating `VirtualMachines` across several different deployments.
+
+The deployment of the cluster-wide `kubevirt/common-instancetypes` resources by
+default will allow users to rely more freely on the presence of the resources
+and make use of them during their `VirtualMachine` creation flows.
+
+For example, thanks to recent improvements within `virtctl` the following
+command would deploy a simple Fedora `VirtualMachine` work across several
+different deployments:
+
+```sh
+$ virtctl create vm \
+  --instancetype n1.medium \
+  --preference fedora \
+  --volume-containerdisk name:fedora,src:quay.io/containerdisks/fedora:latest \
+  --name fedora | kubectl apply -f -
+```
+
+## Goals
+
+* For new deployments, the deployment of the cluster-wide
+  `kubevirt/common-instancetypes` resources by `virt-operator`
+
+* For upgraded deployments, to have `virt-operator` claim ownership of any
+  existing cluster-wide `kubevirt/common-instancetypes` resources deployed by
+  the SSP operator.
+
+## Non Goals
+
+* For upgraded deployments, `virt-operator` will not try to claim ownership of
+  any operator deployed cluster-wide `kubevirt/common-instancetypes` resources
+  other than by SSP.
+
+## User Stories
+
+* As a user I want access to a consistent default set of cluster-wide instance
+  types and preferences across all deployments I interact with
+
+* As an operator I want my users to have access to a consistent set of
+  cluster-wide instance types and preferences without the need for additional
+  deployment of operators or resources
+
+## Repos
+
+* kubevirt/kubevirt
+* kubevirt/common-instancetypes
+* kubevirt/ssp-operator
+
+# Design
+
+`virt-operator` will be extended to deploy bundles of cluster-wide resources by
+default from the `kubevirt/common-instancetypes` project.
+
+These bundles will be injected into the `virt-operator` container image at build
+time with the `kubevirt/common-instancetypes` project referenced as a submodule
+checked out from a specific release tag from within `kubevirt/kubevirt`.
+
+The submodule approach keeps much of the maintenance and testing cruft of the
+`kubevirt/common-instancetypes` project out of the already congested
+`kubevirt/kubevirt` project. It also allows us to retain the existing
+`kustomize` project structure of `kubevirt/common-instancetypes` making the repo
+useful for end operators and users looking to make modifications to or to only
+generate a subset of the resources.
+
```

Comment:
@davidvossel @fabiand I've tried to set out my reasoning for using a sub-module above, let me know if this is still an issue and I can think about reworking things into a flat import of these resources into `kubevirt/kubevirt`.